src/dimension.jl

```diff
@@ -212,9 +216,27 @@ dimmacro(typ, supertype, name=string(typ), shortname=string(typ)) =
 
 @dim Ti TimeDim "Time"
 @doc """
-Time dimension. `Ti <: TimeDim <: IndependentDim 
+Time dimension. `Ti <: TimeDim <: IndependentDim
 
 `Time` is already used by Dates, so we use `Ti` to avoid clashing.
 """ Ti
 
 const Time = Ti # For some backwards compat
+
+#########################################################################
+# coordinate dimensions
+#########################################################################
+struct Coord{T<:Vector{<:AbstractVector}, G, M, D} <: AbstractDimension{T,G,M}
+    val::T
+    dims::Tuple{D}
```

Comment:
Instead of `Tuple{D}`, just use `D`, and put `D <: Tuple` in the definition above.